### PR TITLE
feat(opentelemetry-instrumentation-aws-lambda): support otel links in SQS and lambda triggers

### PR DESCRIPTION
**Pull request by NimrodAvni78**
## Which problem is this PR solving?
adding links from lambda A to lambda B in the following scenarios
lambda A produce SQS message -> SQS -> lambda B triggered by SQS
lambda A produce event -> Event Bridge -> lambda B triggered by event

## Short description of the changes
for SQS, we propagate context under the message attributes
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html

image

for EventBridge, modify the message under details, adding the `_context` key where we propagate the context in
https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html

image

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts [hunk 1]
```diff
@@ -29,17 +42,71 @@ const isEventBridgeEvent = (
...
+      `EventBridge instrumentation: cannot extract propagated context from EventBridge event due to event Detail is not a valid json: ${entry.detail}`
+    );
+    return undefined;
+  }
+};
+
+const extractEventBridgeLink = (
+  event: EventBridgeEvent<string, any>
+): Link | undefined => {
+  const parsedDetails = getDetailsFromEvent(event);
```
Comment (NimrodAvni78) [inline plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:72]: TODO maybe we need to delete the _context key from the event after we extract the context from it

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts [hunk 2]
```diff
@@ -29,17 +42,71 @@ const isEventBridgeEvent = (
...
+  if (!context) return undefined;
+  return { context };
+};
+
 function initializeEventBridgeSpan(
   event: EventBridgeEvent<string, any>
 ): TriggerSpanInitializerResult {
   const attributes: Attributes = {
-    'aws.event.bridge.trigger.service': event.source,
+    'aws.event.bridge.trigger.source': event.source,
```
Comment (galrose) [inline plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:99]: Will this change the attributes in the spans for the lambdas? If so is it aligned with python and ill need to do a fix in ws-serverless

Comment (NimrodAvni78) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:99]: ohhh i didnt know anyone relies on this,
i can revert this there is no real need

Comment (galrose) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:99]: I'll check in the serverless soon and let u know

Comment (NimrodAvni78) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:99]: i changed it just because i think its a better name

Comment (galrose) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:99]: https://github.com/search?q=org%3Acoralogix+trigger.service&type=code

its not me but take a look its used in some places

Comment (galrose) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:99]: I think its all u except 1

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts (hunk 1 above)
Comment (galrose) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:72]: why TODO?

Comment (NimrodAvni78) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:72]: more of a "Should we do it?" then a TODO

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts (hunk 2 above)
Comment (NimrodAvni78) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:99]: ok ill revert it back

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts (hunk 1 above)
Comment (galrose) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:72]: it the client has a way to see it then maybe but I dont think it really matters

**Review by galrose — APPROVED** [review]
LGTM

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts (hunk 1 above)
Comment (NimrodAvni78) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:72]: yeah he has access to it, but i think you as a client also have access to HTTP headers used for context propagation and i dont think they are removed, so im not sure how much it matters

Comment (galrose) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/triggers/event-bridge.ts:72]: Shouldn't matter much,up to you

**Pull request merged**